compiler/luci/pass/src/CircleShapeInferencepass.test.cpp

```diff
@@ -0,0 +1,356 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/CircleShapeInferencePass.h"
+
+#include <loco.h>
+
+#include <luci/IR/CircleNodes.h>
+
+#include <gtest/gtest.h>
+
+/**
+ * This test is for checking when traversal order affect the result of inference.
+ *
+ * Because "transpose1" only changed permutation node from "old_perm" to "new_perm",
```

Comment:
>  Because "transpose1" only changed permutation node from "old_perm" to "new_perm",

This line is understood as "transpose1" will change permutation node by itself
- really?
- there is no when or why

Reply:
Oops...? Then how about
`When perm() of "transpose1" is changed from "old_perm" to "new_perm" by some of luci/Pass,` ?

Reply:
That's better :)

`When perm() of "transpose1" is changed from "old_perm" to "new_perm" by some of luci/Pass in below diagram,`